pkg/deploy/lattice/targets_manager.go

```diff
@@ -111,23 +106,61 @@ func (s *defaultTargetsManager) deregisterStaleTargets(
 				break
 			}
 		}
-
 		if isStale {
 			targetsToDeregister = append(targetsToDeregister, &vpclattice.Target{Id: latticeTarget.Id, Port: latticeTarget.Port})
 		}
 	}
-
-	if len(targetsToDeregister) > 0 {
+	if len(targetsToDeregister) == 0 {
+		return nil
+	}
+	partitionedTargetsToDeregister := getPartitionedLatticeTargets(targetsToDeregister, 100)
```

Comment:
Dont need getPartitionedLatticeTargets either

```go
	const chunkSize = 100
	for low := 0; low < len(targetsToDeregister); low += chunkSize {
		high := min(low+chunkSize, len(targetsToDeregister))
		chunk := targetsToDeregister[low:high]
	}
```

Unfortunately we still dont have min function in Go as of 1.20, but it's coming with 1.21 (any other sane language has it). You can add this one too.

```go
// built-in from golang 1.21 (we are at 1.20)
func min[T constraints.Ordered](a, b T) T {
	if a < b {
		return a
	}
	return b
}
```

Reply:
I could change to this style, but still think put Partition logic in a separate function make it clearer